bioconda_utils/cli.py

```diff
@@ -578,11 +583,13 @@ def update_pinning(recipe_folder, config, packages="*",
     hadErrors = set()
     bumpErrors = set()
 
-    needs_bump = partial(update_pinnings.check, build_config=build_config)
+    needs_bump = partial(
+        update_pinnings.check, build_config=build_config, skip_variant_keys=skip_variants,
+    )
 
     State = update_pinnings.State
 
-    for status, recip in zip(utils.parallel_iter(needs_bump, dag, "Processing..."), dag):
+    for status, recip in utils.parallel_iter(needs_bump, dag, "Processing..."):
```

Comment:
The order of `utils.parallel_iter(...)` and `iter(dag)` is different, apparently, which means we got "random" misassigned build bumps.